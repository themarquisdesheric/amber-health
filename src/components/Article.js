import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import PreviewCompatibleImage from './PreviewCompatibleImage';

const Article = ({ post, className = '' }) => (
  <article className={`${className} mb-8`}>
    <Link to={post.fields.slug}>
      <header>
        {post.frontmatter.featuredimage ? (
          <div className="w-full">
            <PreviewCompatibleImage
              imageInfo={{
                image: post.frontmatter.featuredimage,
                alt: `featured image thumbnail for post ${post.title}`
              }}
            />
          </div>
        ) : null}
        <p className="text-lg font-bold pt-5 pb-2">
          {post.frontmatter.title}
        </p>
      </header>
      <p className="text-sm pb-2">{post.frontmatter.description}</p>
    </Link>
  </article>
);

Article.propTypes = {
  post: PropTypes.object.isRequired,
  className: PropTypes.string
};

export default Article;
