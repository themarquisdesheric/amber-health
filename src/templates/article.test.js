import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Article from './article';

const data = {
  markdownRemark: {
    id: '73358e5e-d586-563b-8e98-34c40eaf71f2',
    html: '<p>some markup</p>',
    frontmatter: {
      date: 'September 28, 2019',
      title: 'Endometriosis Treatments',
      description: 'Learn about the available treatments for endometriosis and the importance of a multidisciplinary approach to disease management ',
      tags: [
        'endometriosis',
        'women\'s health'
      ]
    }
  }
};

test('Article', () => {
  const wrapper = shallow(<Article data={data} />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
