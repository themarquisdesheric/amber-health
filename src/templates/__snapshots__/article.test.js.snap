// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article 1`] = `
<TemplateWrapper>
  <ArticleTemplate
    content="<p>some markup</p>"
    contentComponent={[Function]}
    description="Learn about the available treatments for endometriosis and the importance of a multidisciplinary approach to disease management "
    helmet={
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
        titleTemplate="%s | The Chronic"
      >
        <title>
          Endometriosis Treatments
        </title>
        <meta
          content="Learn about the available treatments for endometriosis and the importance of a multidisciplinary approach to disease management "
          name="description"
        />
        <meta
          content="article"
          property="og:type"
        />
        <meta
          content="Endometriosis Treatments"
          property="og:title"
        />
        <meta
          content="Learn about the available treatments for endometriosis and the importance of a multidisciplinary approach to disease management "
          property="og:description"
        />
        <meta
          content="/endometriosis-treatments"
          property="og:url"
        />
        <meta
          content="/img/sonjiawpom-1.jpg"
          property="og:image"
        />
        <meta
          content="summary"
          name="twitter:card"
        />
        <meta
          content="@info_chronic"
          name="twitter:site"
        />
        <meta
          content="@info_chronic"
          name="twitter:creator"
        />
        <meta
          content="/endometriosis-treatments"
          name="twitter:url"
        />
        <meta
          content="Endometriosis Treatments"
          name="twitter:title"
        />
        <meta
          content="Learn about the available treatments for endometriosis and the importance of a multidisciplinary approach to disease management "
          name="twitter:description"
        />
        <meta
          content="/img/sonjiawpom-1.jpg"
          name="twitter:image"
        />
        <meta
          content="woman holding pomegranate"
          name="twitter:image:alt"
        />
        <meta
          content="endometriosis, women's health, "
          name="keywords"
        />
      </HelmetWrapper>
    }
    tags={
      Array [
        "endometriosis",
        "women's health",
      ]
    }
    title="Endometriosis Treatments"
  />
</TemplateWrapper>
`;
